types/node.lua

```diff
@@ -204,17 +177,34 @@ function nodetypes.hbox:outputYourself (typesetter, line)
    end
 end
 
-nodetypes.zerohbox = pl.class(nodetypes.hbox)
-nodetypes.zerohbox.type = "zerohbox"
-nodetypes.zerohbox.value = { glyph = 0 }
-
-nodetypes.nnode = pl.class(nodetypes.hbox)
-nodetypes.nnode.type = "nnode"
-nodetypes.nnode.language = ""
-nodetypes.nnode.pal = nil
-nodetypes.nnode.nodes = {}
-
-function nodetypes.nnode:_init (spec)
+--- A zerohbox is a special-kind of hbox with zero width, height and depth.
+--
+-- Derived from `hbox`.
+--
+-- Properties is_zerohbox (and convenience is_zero) and is_box are true.
+-- Note that is_hbox is NOT true: zerohbox are used in a specific context
+--
+-- @type zerohbox
+local zerohbox = pl.class(hbox)
+zerohbox.type = "zerohbox"
+zerohbox.value = { glyph = 0 }
```

Comment:
I wondered what this standalone use of `glyph` in this module here.

I _think_ it comes from the pango shaper (regex-searched for occurrences, so maybe I haven't been exhaustive): It's `shaper:shapeToken` does return a "glyph" (vs. value.glyphString etc. with harfbuzz).

But I've a **serious** concern here. Nodes ought to be shaper-agnostic, at least at some level. Or is every potential shaper implementation put its crap in the nodes? And how can we expect using them correctly in packages etc. then?

Reply:
This is exactly why I've been nursing along the Pango shaper at various points just to verify that our shaper API is sane. I don't know anything about this particular case, but in order to be shaper agnostic our nodes do need enough information attached to please any reasonably sane shaper, no?

If we were to drop down to just one shaper our API surface area would be smaller, but SILE itself would be *far* less robust.